uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-api/src/main/java/org/uberfire/ext/editor/commons/service/htmleditor/HtmlEditorService.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.uberfire.ext.editor.commons.service.htmleditor;
+
+import org.jboss.errai.bus.server.annotations.Remote;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.ext.editor.commons.file.DefaultMetadata;
+import org.uberfire.ext.editor.commons.service.support.SupportsCopy;
+import org.uberfire.ext.editor.commons.service.support.SupportsDelete;
+import org.uberfire.ext.editor.commons.service.support.SupportsRename;
+import org.uberfire.ext.editor.commons.service.support.SupportsSaveAndRename;
+
+@Remote
+public interface HtmlEditorService extends SupportsCopy,
+                                           SupportsRename,
+                                           SupportsDelete,
+                                           SupportsSaveAndRename<String, DefaultMetadata> {
+
+    String readAllString(final Path path);
+
+    Path write(final Path path, final String content);
```

Comment:
Same question about ```SupportsUpdate<String, DefaultMetadata>```?

Reply:
Ah.. OK, so is this needed at all as you're extending ```SupportsSaveAndRename``` that extends ```SupportsUpdate```?

Reply:
Done.